### PR TITLE
fix:fix the manual delete servers.pid file after restart container

### PR DESCRIPTION
**Pull request by RayGuo-ergou**
# Description

When we restart the doubtfire-api container, we need to delete the ` server.pid ` file in ` /tmp/pids ` manually. So I added a docker entrypoint srcipt to delete that file everytime before the container start if  ` server.pid ` exist. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix

# How Has This Been Tested?

I build the docker image in my local dev environment, then develop the project with that image. To test it, I firstly went to the container bash, make sure the ` server.pid ` file is in the right path then I restart the container several times and restart my PC to test if we still need to delete ` server.pid ` file manually. Also by monitoring the pids folder I can clearly see the file was removed then created when restart the container. 

- [x] restart container
- [x] reboot PC

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation if appropriate
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have created or extended unit tests to address my new additions
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by RayGuo-ergou** [comment]
Ref: [https://stackoverflow.com/questions/35022428/rails-server-is-still-running-in-a-new-opened-docker-container](https://stackoverflow.com/questions/35022428/rails-server-is-still-running-in-a-new-opened-docker-container), [[entrypoint: "entrypoint.sh" - docker compose](https://stackoverflow.com/questions/58438300/entrypoint-entrypoint-sh-docker-compose)](https://stackoverflow.com/questions/58438300/entrypoint-entrypoint-sh-docker-compose)

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+if [ -f tmp/pids/server.pid ]; then
+  rm tmp/pids/server.pid
+fi
```
Comment (jakerenzella) [inline docker-entrypoint.sh:6]: This could just be:

`rm -f tmp/pids/server.pid`

**Comment by jakerenzella** [comment]
Nice! This annoys me all the time

File: docker-entrypoint.sh (hunk 1 above)
Comment (RayGuo-ergou) [reply docker-entrypoint.sh]: This @jakerenzella for you suggestion. It's done.

**Comment by macite** [comment]
@RayGuo-ergou I dont think the **bundle exec** is needed in the _docker-entrypoint.sh_ as the CMD has this already. This will then also work if you want to run things outside of bundler.

**Comment by RayGuo-ergou** [comment]
Hi @macite I have removed ` bundle exec ` in the *docker-entrypoint.sh* file.

**Comment by jakerenzella** [comment]
Do we need the entry point file at all then?

**Comment by RayGuo-ergou** [comment]
@jakerenzella Good point, maybe we can just add the ` rm -f  ` command into CMD?

**Comment by jakerenzella** [comment]
> @jakerenzella Good point, maybe we can just add the ` rm -f  ` command into CMD?

Yep 👍 keep it simple

**Comment by jakerenzella** [comment]
@RayGuo-ergou  this looks good but just need to double check that rm -f always returns a truthy value even if there's no file, otherwise the proceeding statements won't run

**Comment by jakerenzella** [comment]
If not you can do rm -rf || command2 && command3 etc

**Comment by RayGuo-ergou** [comment]
> @RayGuo-ergou this looks good but just need to double check that rm -f always returns a truthy value even if there's no file, otherwise the proceeding statements won't run

I tested to start the container that has no *server,pid* file and it's running okay. Is that what you mean, sorry I am not very sure.

**Comment by jakerenzella** [comment]
> > @RayGuo-ergou this looks good but just need to double check that rm -f always returns a truthy value even if there's no file, otherwise the proceeding statements won't run
> 
> I tested to start the container when the _server,pid_ file is deleted and it's running okay. Is that what you mean, sorry I am not very sure.
> 
> `ls` can be excuted image

This looks all good then :)

**Pull request merged**